certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -332,9 +333,12 @@ def ipv6_info(self, port):
                     ipv6only_present = True
         return (ipv6_active, ipv6only_present)
 
-    def _vhost_from_duplicated_default(self, domain, port=None):
+    def _vhost_from_duplicated_default(self, domain, allow_port_mismatch, port=None):
```

Comment:
Can we require `port` as a param now? Or is there a situation where we might want to unambiguously select a single default vhost and ignore the port entirely?

Reply:
only in theory not in practice, and since we never in practice call in without a port, happy to make it not optional